src/java/org/apache/cassandra/config/GuardrailsOptions.java

```diff
@@ -122,48 +135,59 @@ public boolean getUserTimestampsEnabled()
 
     public void setUserTimestampsEnabled(boolean enabled)
     {
-        user_timestamps_enabled = enabled;
+        updatePropertyWithLogging(NAME_PREFIX + "user_timestamps_enabled",
+                                  enabled,
+                                  () -> user_timestamps_enabled,
+                                  x -> user_timestamps_enabled = x);
     }
 
-    public static abstract class Threshold implements org.apache.cassandra.db.guardrails.Threshold.Config
+    private static <T> void updatePropertyWithLogging(String name, T value, Supplier<T> getter, Consumer<T> setter)
     {
-        public static final long DISABLED = -1;
-
-        public volatile long warn_threshold = DISABLED;
-        public volatile long abort_threshold = DISABLED;
-
-        @Override
-        public long getWarnThreshold()
+        T previous = getter.get();
+        if (!value.equals(previous))
         {
-            return warn_threshold;
+            setter.accept(value);
+            logger.info("updated {} from {} to {}", name, previous, value);
         }
+    }
 
-        @Override
-        public long getAbortThreshold()
-        {
-            return abort_threshold;
-        }
+    protected static abstract class Config
+    {
+        protected String name;
 
-        public void setThresholds(long warn, long abort)
+        public String getName()
         {
-            validateStrictlyPositive(warn, "warn threshold");
-            validateStrictlyPositive(abort, "abort threshold");
-            validateWarnLowerThanAbort(warn, abort, null);
-            warn_threshold = warn;
-            abort_threshold = abort;
+            return name;
         }
 
-        public void validate(String name)
+        protected void validate(String name)
         {
-            validateStrictlyPositive(warn_threshold, name + ".warn_threshold");
-            validateStrictlyPositive(abort_threshold, name + ".abort_threshold");
-            validateWarnLowerThanAbort(warn_threshold, abort_threshold, name);
```

Comment:
I think it was fine to have this check in a separate method as the others, but I am fine also to leave it as is now.

Reply:
I have put it back into a separate method. I inlined that method because there wasn't any other caller, but maybe it's clearer if we keep that packed into a method. Also, we might end up having a separate caller if we add separate setter for warn/abort in the future.